### PR TITLE
update walkthrough.md

### PR DESCRIPTION
**Pull request by mengqiy**
Update the `walkthrough.md` doc.
Mention how to setup the firewall rules for cloud provider.

When I first try this walkthrough, `kubectl` failed to talk to service catalog API server. But `kubectl` just said something timeout, which is not helpful at all.
So I think include this part will be helpful.

### CODE REVIEW

**Comment by MHBauer** [comment]
I don't like having this in the main flow. Can we put it in some platform note section at the end? With maybe a note at the start to "see the troubleshooting section"?

Whose duty is it to know what platform-specific thing is needed? 

Does something need to be done for Bluemix, Azure, AWS, etc?

**Comment by mengqiy** [comment]
> I don't like having this in the main flow. Can we put it in some platform note section at the end? With maybe a note at the start to "see the troubleshooting section"?

That's fair. I will move it.

> Whose duty is it to know what platform-specific thing is needed?
Does something need to be done for Bluemix, Azure, AWS, etc?

I just want to remind people and save their time. If you think the part I'm adding is too specific. I can delete the GCP specific commands, and change it to something like
```
You may need to config your firewall rules to allow incoming traffic in if you are running service catalog with a cloud provider.
Please refer your cloud provider's documents.
```

File: docs/walkthrough.md [hunk 1]
```diff
@@ -193,6 +193,15 @@ kubectl config set-cluster service-catalog --server=http://$SVC_CAT_API_SERVER_I
 kubectl config set-context service-catalog --cluster=service-catalog
 ```
 
+Note: if you are using Google cloud or other cloud providers, you may need to
+setup proper firewall rules to allow your traffic get in.
+On Google cloud, you need to run the following commands:
+
+```console
+gcloud compute firewall-rules create allow-service-catalog-insecure --allow tcp:30080 --description "Allow incoming traffic on 30080 port."
```
Comment (MHBauer) [inline docs/walkthrough.md]: Insecure is about to go away. I think several steps in here are going to need to be edited based on that. Not blocking, just a note.

File: docs/walkthrough.md [hunk 2]
```diff
@@ -193,6 +193,15 @@ kubectl config set-cluster service-catalog --server=http://$SVC_CAT_API_SERVER_I
 kubectl config set-context service-catalog --cluster=service-catalog
 ```
 
+Note: if you are using Google cloud or other cloud providers, you may need to
+setup proper firewall rules to allow your traffic get in.
+On Google cloud, you need to run the following commands:
```
Comment (MHBauer) [inline docs/walkthrough.md]: Move this and the following to a new toplevel `#troubleshooting` section at the end.

File: docs/walkthrough.md [hunk 3]
```diff
@@ -193,6 +193,15 @@ kubectl config set-cluster service-catalog --server=http://$SVC_CAT_API_SERVER_I
 kubectl config set-context service-catalog --cluster=service-catalog
 ```
 
+Note: if you are using Google cloud or other cloud providers, you may need to
```
Comment (MHBauer) [inline docs/walkthrough.md]: I like the note here.  "Your cloud provider may require firewall rules to allow your traffic get in."

I think we should move the following details on particular cloud providers to a new section at the end, and link there from here.

**Comment by kibbles-n-bytes** [comment]
I agree with @MHBauer ; I like having just the note cautioning the user. We can have troubleshooting info at the bottom.

Also, if you're using a GCP project under the "google.com" organization, I believe creating a firewall rule isn't enough, at least for the insecure case. I tried doing that, but it seems GCP regularly deletes them without warning due to security concerns.

File: docs/walkthrough.md (hunk 3 above)
Comment (pmorie) [inline docs/walkthrough.md]: I'm +1 on a cloud-provider-errata section.

**Comment by mengqiy** [comment]
GCP deny all ingress traffic by default no matter what kind of project it is. See: https://cloud.google.com/compute/docs/vpc/firewalls#default_firewall_rules

> Also, if you're using a GCP project under the "google.com" organization, I believe creating a firewall rule isn't enough, at least for the insecure case. I tried doing that, but it seems GCP regularly deletes them without warning due to security concerns.

That's a totally different story. I can chat you about this.

**Comment by mengqiy** [comment]
I will update the PR to restructure it soon.

**Comment by pmorie** [comment]
Thanks for the submission @mengqiy

**Comment by mengqiy** [comment]
Updated. PTAL

**Comment by MHBauer** [comment]
@mengqiy anything else? Taking out of in-progress for now.

**Comment by mengqiy** [comment]
> @mengqiy anything else?

@MHBauer What else are you expecting? Did I miss something?

**Comment by MHBauer** [comment]
Nothing from me. Wasn't sure if the inprogress tag was still relevant.

File: docs/walkthrough.md [hunk 4]
```diff
@@ -205,6 +205,9 @@ kubectl config set-cluster service-catalog --server=https://$SVC_CAT_API_SERVER_
 kubectl config set-context service-catalog --cluster=service-catalog
 ```
 
+Note: Your cloud provider may require firewall rules to allow your traffic get in.
+Please refer the [Troubleshooting](#troubleshooting) section for details.
```
Comment (kibbles-n-bytes) [inline docs/walkthrough.md]: "refer the" -> "refer to the"

Comment (kibbles-n-bytes) [inline docs/walkthrough.md]: "refer the" -> "refer to the"

File: docs/walkthrough.md [hunk 5]
```diff
@@ -502,3 +505,14 @@ Then, delete all the namespaces we created:
 ```console
 kubectl delete ns test-ns catalog ups-broker
 ```
+
+## Troubleshooting
+
+### Firewall rules
+
+If you are using Google Cloud Platform, you may need to run the following
+commands setup proper firewall rules to allow your traffic get in.
```
Comment (kibbles-n-bytes) [inline docs/walkthrough.md]: "commands setup" -> "command to setup"

**Comment by mengqiy** [comment]
Fixed grammatical nits.

**Comment by duglin** [comment]
LGTM

**Pull request merged**